src/Digdir.Domain.Dialogporten.Domain/Dialogs/Entities/DialogEntity.cs

```diff
@@ -89,17 +91,48 @@ public void OnDelete(AggregateNode self, DateTimeOffset utcNow)
         _domainEvents.Add(new DialogDeletedDomainEvent(Id, ServiceResource, Party));
     }
 
-    public void UpdateReadAt(DateTimeOffset timestamp)
+    public void UpdateSeenAt(string seenByEndUserId, string seenByEndUserName)
     {
-        if ((ReadAt ?? DateTimeOffset.MinValue) >= UpdatedAt)
+        var lastSeenByAt = Activities
+            .Where(x => x.SeenByEndUserId == seenByEndUserId)
+            .MaxBy(x => x.CreatedAt)
+            ?.CreatedAt;
+
+        if ((lastSeenByAt ?? DateTimeOffset.MinValue) >= UpdatedAt)
         {
             return;
         }
 
-        ReadAt = timestamp;
-        _domainEvents.Add(new DialogReadDomainEvent(Id));
+        if (string.IsNullOrWhiteSpace(seenByEndUserName))
+        {
+            seenByEndUserName = "NavnIkkeFunnet";
+        }
+
+        Activities.Add(new DialogActivity
+        {
+            Description = new DialogActivityDescription
```

Comment:
Fjern description, kan utledes a TypeId. Det som faktisk vises bestemmes GUI.

Reply:
Den er lagt til fordi description er p.t. et "påkrevd" felt. Altså "en aktivitet har 1 til mange descriptions". Så for å være konsekvent la vi denne inn også.

Reply:
Er det bare håndhevd av CreateDialogDialogActivityDtoValidator / UpdateDialogDialogActivityDtoValidator, eller EF? Synes ikke det gir mening å ha en beskrivelse her. Kan vi endre dette kravet til å kun gjelde hvis Type ikke er Seen? 

Kommer på at vi uansett må endre validatoren slik at TE ikke kan legge inn Seen- eller Forwarded-oppslag (de skal kun komme fra "oss").

Reply:
Det er ikke håndhevet i EF nei. Kun i applikasjonslag validator. Det kan bli en WTF faktor for konsumentene av Dialogporten. "Alle aktiviteter må inneholde minst en description, sett bort i fra de dialogporten selv lager". 

Vi sier at TE er produsent av dataene (de valideres), men så har vi unntak hvor DP selv produserer data, og for de unntakene er det enda flere unntak. Det kan fort bli en WTF faktor, både for TE og SB - og selv for oss en gang i fremtiden ser jeg for meg 😅 Men jeg forstår hva du sier konseptuelt. Det gir ikke mening å ikke ha descriotion på seen aktiviteter. Jeg liker dog ikke denne unntaket for unntaket for unntaket - if you know what I mean. Føles nesten at vi går i mot Liskov Substitution Principle.

Dersom vi ikke skal ha description på "våre" aktiviteter, syns jeg heller ikke vi skal kreve det på TE sine. Men skal ikke være religiøs på det :grin: